### PR TITLE
[WIP] Add support for init of a collection

### PR DESCRIPTION
**Pull request by jctanner**
This is the newstyle directory layout for content going into galaxy. We've decided to call them "collections", and this pr add an init subcommand to make them.

The layout spec comes from https://github.com/bcoca/collection
Some preliminary examples: https://github.com/ansible-collections/
How we'll eventually install these to the clients: https://galaxy.ansible.com/docs/mazer/index.html
Outdated support for ansible to consume these https://github.com/ansible/ansible/tree/mazer_role_loader

#### PR Type

- Feature Pull Request

### CODE REVIEW

**Comment by webknjaz** [comment]
@jctanner `tox -e format` will fix most of the linter offenses.

File: molecule/command/init/collection.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@click.command()
+@click.pass_context
+@click.option(
+    '--collection-namespace', '-n', required=True, help='Namespace of the collection.')
+@click.option(
+    '--collection-name', '-c', required=True, help='Name of the collection.')
+def collection(ctx, collection_namespace, collection_name): #pragma: no cover
+    """ Initialize a new collection for use with Molecule. """
```
Comment (webknjaz) [inline molecule/command/init/collection.py]: Per PEP 257, you should strip spaces

File: molecule/command/init/collection.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        LOG.success(msg)
+
+
+@click.command()
+@click.pass_context
+@click.option(
+    '--collection-namespace', '-n', required=True, help='Namespace of the collection.')
+@click.option(
+    '--collection-name', '-c', required=True, help='Name of the collection.')
+def collection(ctx, collection_namespace, collection_name): #pragma: no cover
```
Comment (webknjaz) [inline molecule/command/init/collection.py]: PEP8: two spaces needed before hash and one after it

File: molecule/cookiecutter/collection/{{cookiecutter.collection_name}}/galaxy.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+##################################################
```
Comment (webknjaz) [inline molecule/cookiecutter/collection/{{cookiecutter.collection_name}}/galaxy.yml]: JFYI you can use https://pytest-cookies.readthedocs.io/ to test cookiecutter template rendering.

File: molecule/command/init/collection.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+#  Copyright (c) 2015-2018 Cisco Systems, Inc.
```
[the remark this answers is not in the document]
Comment (bcoca) [reply molecule/command/init/collection.py]: 'molecule project'?

**Comment by jctanner** [comment]
We have more work to do on this before it's ready for molecule to consume.

**Comment by decentral1se** [comment]
I'm just nagging on #ansible-galaxy chat now but if there is any further help I can offer on this, I'd be glad to! Would be very cool to get mazer hooked into molecule ASAP to get playing around with collections. Testing roles with multiple roles is one of the main topics that keeps coming up in molecule requests and discussions.

File: molecule/cookiecutter/collection/{{cookiecutter.collection_name}}/galaxy.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# short summary of the collection
+description:
+# a valid SPDX license identifier https://spdx.org/licenses/
+license:
+# list of keywords you want to associate the collection with for indexing/search systems
+tags:
+# list of dependencies, other collections this collection requires to be installed for it to be usable
+dependencies:
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply molecule/cookiecutter/collection/{{cookiecutter.collection_name}}/galaxy.yml]: The empty list should be fine I think

**Pull request closed without merging**